app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,19 @@
+class LikesController < ApplicationController
+  def create
+    @user = current_user
+    @post = Post.find(params[:post_id])
+    @like = @post.likes.new(like_params)
+
+    respond_to do |format|
+      if @like.save
+        format.html { redirect_to user_post_path(@post.author, @post), notice: 'Like was successfully created.' }
+      end
+    end
+  end
```

Comment:
- [ ]  Looking at the video below, you can see that I get an error when I try to like a post.  The error seems to be referring to `param is missing or the value is empty`. You can fix this by removing the `require(:like)` you have in the like param, having only this `params.permit(:post_id)`. :+1: Also The format you are returning in the likes create method is not correct and It will throw an error when you try to create a like. I'll suggest you fix these and test your app to make sure all works as expected before requesting a re-review. :+1: :100: 

- - Video Reference:

https://github.com/Abdulwhbhussain/blog-app/assets/107632780/a085c0f7-abbd-4130-9346-9f20a298c0c9

Reply:
Fixed Bro!